src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -1041,6 +1053,14 @@ std::pair< bool, nuraft::cb_func::ReturnCode > RaftReplDev::handle_raft_event(nu
                 entries.size(), raft_req->get_last_log_term(), start_lsn, start_lsn + entries.size() - 1,
                 m_commit_upto_lsn.load(), raft_req->get_commit_idx());
 
+        if (start_lsn != m_data_journal->next_slot()) {
+            // if the start_lsn of this batch does not match the next_slot, drop this log batch
+            // this will happen when the leader is sending the logs which are already received and appended
+            RD_LOGD("start_lsn={} does not match the next_slot={}, dropping the log batch", start_lsn,
```

Comment:
Please add more comments here why we want to skip append on this case.

Reply:
you are right , this part should be remove , nuraft will handle all the cases